frontend/src/pages/modelServing/customServingRuntimes/CustomServingRuntimeListView.tsx

```diff
@@ -1,60 +1,56 @@
 import { useDashboardNamespace } from '~/redux/selectors';
 import useNotification from '~/utilities/useNotification';
-import { compareTemplateKinds } from './utils';
+import { getDragItemOrder } from './utils';
+import DeleteCustomServingRuntimeModal from './DeleteCustomServingRuntimeModal';
 import { columns } from './templatedData';
 import CustomServingRuntimeTableRow from './CustomServingRuntimeTableRow';
+import { CustomServingRuntimeContext } from './CustomServingRuntimeContext';
 
-type CustomServingRuntimeListViewProps = {
-  templates: TemplateKind[];
-  templateOrder: string[];
-  refreshOrder: () => void;
-};
-
-const CustomServingRuntimeListView: React.FC<CustomServingRuntimeListViewProps> = ({
-  templates,
-  templateOrder,
-  refreshOrder,
-}) => {
+const CustomServingRuntimeListView: React.FC = () => {
+  const {
+    servingRuntimeTemplateOrder: { data: templateOrder },
+    servingRuntimeTemplates: { data: templates },
+    refreshData,
+  } = React.useContext(CustomServingRuntimeContext);
   const { dashboardNamespace } = useDashboardNamespace();
-  const initialItemOrder = React.useMemo(
-    () =>
-      templates.sort(compareTemplateKinds(templateOrder)).map((template) => template.metadata.name),
-    [templates, templateOrder],
-  );
   const notification = useNotification();
   const navigate = useNavigate();
-
-  const onDropCallback = React.useCallback(
-    (newTemplateOrder) => {
-      if (!_.isEqual(newTemplateOrder, templateOrder)) {
-        patchDashboardConfigTemplateOrder(newTemplateOrder, dashboardNamespace)
-          .then(refreshOrder)
-          .catch((e) => notification.error(`Error update the serving runtimes order`, e.message));
-      }
-    },
-    [templateOrder, dashboardNamespace, refreshOrder, notification],
+  const [deleteTemplate, setDeleteTemplate] = React.useState<TemplateKind>();
+  const [dragItemOrder, setDragItemOrder] = React.useState(
+    getDragItemOrder(templates, templateOrder),
   );
 
+  React.useEffect(() => {
+    patchDashboardConfigTemplateOrder(dragItemOrder, dashboardNamespace).catch((e) =>
+      notification.error(`Error update the serving runtimes order`, e.message),
+    );
+  }, [dragItemOrder, dashboardNamespace, notification]);
+
+  React.useEffect(() => {
+    setDragItemOrder(getDragItemOrder(templates, templateOrder));
+  }, [templates, templateOrder]);
+
   return (
     <>
       <Table
         isDraggable
         data={templates}
         columns={columns}
-        initialItemOrder={initialItemOrder}
+        itemOrder={dragItemOrder}
+        setItemOrder={setDragItemOrder}
         rowRenderer={(template, rowIndex, trDragFunctions) => (
           <CustomServingRuntimeTableRow
             key={template.metadata.uid}
             obj={template}
             rowIndex={rowIndex}
             dragFunctions={trDragFunctions}
+            onDeleteTemplate={(obj) => setDeleteTemplate(obj)}
           />
         )}
```

Comment:
```suggestion
        rowRenderer={(template, rowIndex, rowId, trDragFunctions) => (
          <CustomServingRuntimeTableRow
            key={template.metadata.uid}
            obj={template}
            rowIndex={rowIndex}
            rowId={rowId || getServingRuntimeNameFromTemplate(template)}
            dragFunctions={trDragFunctions}
            onDeleteTemplate={(obj) => setDeleteTemplate(obj)}
          />
        )}
```

And for this is adapting table, we can either just have flexibility or just giving that for the drag item, iw as thinking about something like this in `Table.tsx`:

` .map((row, rowIndex) => rowRenderer(row, rowIndex, itemOrder[rowIndex], trDragFunctions))}`